src/Filter/Filter.php

```diff
@@ -96,26 +96,26 @@ public function getFieldOption($name, $default = null)
         return $default;
     }
 
-    public function setFieldOption($name, $value): void
+    public function setFieldOption(string $name, $value): void
     {
         $this->options['field_options'][$name] = $value;
     }
 
-    public function getLabel()
+    public function getLabel(): ?string
     {
         return $this->getOption('label');
     }
 
-    public function setLabel($label): void
+    public function setLabel(string $label): void
     {
         $this->setOption('label', $label);
     }
 
-    public function getFieldName()
+    public function getFieldName(): string
```

Comment:
Should we add phpdoc for the exception ? Or should we return null instead if the interface doesnt allow to throw an exception

Reply:
Add a throw would be consistent with current behavior

Reply:
Im used to https://packagist.org/packages/pepakriz/phpstan-exception-rules which allow @throw only if the interface does.

We should add the @throw tag to the interface and document it, wdyt ?

Reply:
Done